### ISSUE TITLE
Squaring the sum of input Field and array value creates an invalid proof

### ISSUE THREAD

**Issue by rahulghangas** [noir-lang/noir#1045]
### Aim

Squaring a value created by adding an input Field and an array value generates an invalid proof

### Expected behavior

Proof verification passes

### Bug

Proof generated from the following program fails verification
```rust
fn main(input : Field, output: pub Field) {
    let arr = [
        1,
        2
    ];

    let out = input + arr[1];
    constrain output == out * out;
}
```

input : `1`
output : `9`

### Installation method

Compiled from source

### Nargo version

nargo 0.3.2 (git version hash: 38bf5719d1757d39c89ecee0a6653a5d9da29c21, is dirty: false)

### Submission Checklist

- [ ] Once I hit submit, I will assign this issue to the Project Board with the appropriate tags.

**Comment by jfecher** [comment]
@rahulghangas I'm not exactly sure yet. I've verified it is correct all the way up until the final SSA output, so the error is either in our ACIR-gen pass (the very last one) or our backend.

The actual `linear_eval is not 0.` error message originates from the backend, but this may be because we generate malformed ACIR.

**Comment by kevaundray** [comment]
Tracking issue: https://github.com/noir-lang/noir/issues/1252

**Comment by rahulghangas** [comment]
@kevaundray confirming that this has been resolved (checked with `0.5.1`). Thank you!
